python/lsst/the/monster/isolate_and_transform.py

```diff
@@ -0,0 +1,149 @@
+DES bandpasses.
+
+The following steps are performed:
+1. Read in the Gaia shard.
+2. Run the IsolatedStarAssociationTask on the Gaia shard.
+3. Read in each catalog.
+4. Match each catalog with the Gaia DR3 catalog.
+5. For each band, interpolate the DES magnitudes.
+6. Save the catalog shard.
+"""
+
+
+class MatchAndTransform:
+    """Match catalogs to Gaia and transform them to 'the_monster'
+       reference frame.
+
+    Parameters
+    ----------
+    htmid : `int`
+        htm level 7 id of catalog(s).
+    catalog_list : `list`
+        List of RefcatInfo objects for catalogs to transform.
+    write_path_inp : `str`
+        (optional) The path to write the outputs to.
+    gaia_cat_info_class : `RefcatInfo object`
+        (optional) The input Gaia DR3 RefcatInfo object.
+    """
+    GaiaDR3CatInfoClass = GaiaDR3Info
+    testing_mode = False
+
+    def run(self,
+            htmid=None,
+            catalog_list=[GaiaXPInfo, SkyMapperInfo, PS1Info, VSTInfo],
+            write_path_inp=None,
+            gaia_cat_info_class=None
+            ):
+
+        # read in gaiaDR3 cat htmid
+        # Read in the Gaia stars in the htmid.
+        if gaia_cat_info_class is None:
+            gaia_info = self.GaiaDR3CatInfoClass()
+        else:
+            gaia_info = gaia_cat_info_class
+
+        gaia_stars_all = read_stars(gaia_info.path, [htmid], allow_missing=self.testing_mode)
+
+        # isolate the gaia cat
+        gaia_stars = self._remove_neighbors(
+            gaia_stars_all
+        )
+        # loop over other catalogs
+        for cat_info in catalog_list:
+            # catalog_list should be a list of
+            # cat_info = self.CatInfoClass() e.g. gaia cat
+
+            # output columns are target catalog id, gaia id, coordinates,
+            # and the des fluxes
+            outcols = ["id", gaia_info.name + "_id", "coord_ra", "coord_dec"]
+            outcols += [f"decam_{band}_flux_from_{cat_info().name}" for band in cat_info().bands]
+
+            # read in star cat (if it exists)
+            if os.path.isfile(cat_info().path+'/'+str(htmid)+'.fits'):
+                cat_stars = read_stars(cat_info().path, [htmid], allow_missing=self.testing_mode)
+
+                # match with gaia_stars
+                with Matcher(gaia_stars["coord_ra"], gaia_stars["coord_dec"]) as m:
+                    idx, i1, i2, d = m.query_knn(
+                        cat_stars["coord_ra"],
+                        cat_stars["coord_dec"],
+                        distance_upper_bound=0.5/3600.0,
+                        return_indices=True,
+                    )
+                cat_stars = cat_stars[i2]
+                cat_stars.add_column(gaia_stars["id"][i1], name=gaia_info.name + "_id")
+
+                for band in cat_info.bands:
+                    # yaml spline fits are per-band, so loop over bands
+                    # read in spline
+                    filename = os.path.abspath(os.path.dirname(__file__))[:-23]
```

Comment:
This should not be coming from `__file__[:-23]` or anything fragile like this.  You use something like:
```
filename = os.path.join(lsst.utils.getPackageDir("the_monster"), "colorterms", f"{cat_info.name}_to_DES_band_{band}.yaml")
```

Reply:
I don't know if it's worth preloading all of these in the `__init__` but that may be premature optimization.